core/cache/application.go

```diff
@@ -26,20 +30,41 @@ type Application struct {
 
 	details    ApplicationChange
 	configHash string
+	hashCache  *hashCache
+}
+
+// Config returns the current application config.
+func (a *Application) Config() map[string]interface{} {
+	a.mu.Lock()
```

Comment:
Out of interest, why the lock here, the underlying Inc is atomic right?

Reply:
Case of being like that in the model example that I cribbed. I will check it and potentially remove the locking there too.

Reply:
Yeah, these have been changed to return copies who's generation is lock-guarded.